service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java

```diff
@@ -1384,6 +1385,120 @@ void shouldReturnAnEmptyOptionalWhenExpectedOrderIsNotContainedWithinHearingOrde
         }
     }
 
+    @Nested
+    class BuildRespondentStatementDynamicList {
+        @Test
+        void shouldBuildDynamicRespondentStatementListFromRespondents() {
+            List<Element<Respondent>> respondents = createRespondents();
+            CaseData caseData = CaseData.builder().respondents1(respondents).build();
+
+            DynamicList expectedDynamicList = ElementUtils
+                .asDynamicList(respondents, null,
+                    respondent -> respondent.getParty().getFullName());
+
+            assertThat(caseData.buildRespondentStatementDynamicList())
+                .isEqualTo(expectedDynamicList);
+        }
+
+        @Test
+        void shouldBuildDynamicRespondentStatementListWithSelectorPropertyFromRespondents() {
+            UUID selectedRespondentId = randomUUID();
+
+            List<Element<Respondent>> respondents = List.of(
+                element(Respondent.builder()
+                    .party(RespondentParty.builder()
+                        .firstName("Sam")
+                        .lastName("Wilson")
+                        .relationshipToChild("Father")
+                        .build())
+                    .build()),
+                element(selectedRespondentId, Respondent.builder()
+                    .party(RespondentParty.builder()
+                        .firstName("Megan")
+                        .lastName("Hannah")
+                        .relationshipToChild("Mother")
+                        .build())
+                    .build()));
+
+            CaseData caseData = CaseData.builder().respondents1(respondents).build();
+
+            DynamicList expectedDynamicList = ElementUtils
+                .asDynamicList(respondents, null,
+                    respondent -> respondent.getParty().getFullName());
+
+            assertThat(caseData.buildRespondentStatementDynamicList()).isEqualTo(expectedDynamicList);
+        }
+    }
+
+    @Nested
+    class GetRespondentStatementByRespondentId {
+        UUID elementId = randomUUID();
+
+        @Test
+        void shouldReturnRespondentStatementWhenRespondentIdMatches() {
+            Element<RespondentStatement> respondentStatementElementOne
+                = element(RespondentStatement.builder().respondentId(elementId).build());
+
+            List<Element<RespondentStatement>> respondentStatements = List.of(
+                respondentStatementElementOne,
+                element(RespondentStatement.builder().build()));
+
+            CaseData caseData = CaseData.builder().respondentStatements(respondentStatements).build();
+
+            Optional<Element<RespondentStatement>> optionalRespondentStatementElement
+                = caseData.getRespondentStatementByRespondentId(elementId);
+
+            assertThat(optionalRespondentStatementElement).isPresent();
+            assertThat(optionalRespondentStatementElement).contains(respondentStatementElementOne);
```

Comment:
codesmell: can be chained:

assertThat(optionalRespondentStatementElement)
.isPresent()
.contains(respondentStatementElementOne);